File: Java/day20.java
import java.util.*;

public class day20
{
    public static void main(String[] args) {
        // Write your code here
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        StringTokenizer tokens = new StringTokenizer(input, "[!,?.\\_'@ ]+");
        System.out.println(tokens.countTokens());
        while (tokens.hasMoreTokens()) {
            System.out.println(tokens.nextToken());
        }
        
    }
}

File: Scala/day20.scala
import scala.util.matching.Regex

object Solution {
  def main(args: Array[String]) {
    // Write your code here
    val str: String = scala.io.StdIn.readLine()
    val pattern = "[^\\W_]+".r
    val words = pattern.findAllIn(str).toArray
    println(words.size)
    words.foreach(println)
  }
}
